### ISSUE TITLE
cannot import backup

### ISSUE THREAD

**Issue by alxp7a** [elevate/elevate.metabase.tools#26]
Command line :
` docker run --rm -v $PWD/data/:/data -v $PWD/metabase.json:/app/appsettings.json elevate/elevate.metabase.tools Command=metabase-exporter.exe Command=import InputFilename=/data/metabase-state.json DatabaseMapping:2=2
`

error : 
```
Unhandled exception. System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JObject' to type 'Newtonsoft.Json.Linq.JArray'.
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
   at metabase_exporter.MetabaseApi.GetAllDatabaseIds() in /sln/metabase-exporter/MetabaseApi.cs:line 243
   at metabase_exporter.MetabaseApiImport.ValidateTargetDatabaseMapping(MetabaseApi api, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 71
   at metabase_exporter.MetabaseApiImport.ValidateDatabaseMapping(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 83
   at metabase_exporter.MetabaseApiImport.Import(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 24
   at metabase_exporter.Program.Import(MetabaseApi api, Import import) in /sln/metabase-exporter/Program.cs:line 45
   at metabase_exporter.Program.Main(String[] args) in /sln/metabase-exporter/Program.cs:line 27
   at metabase_exporter.Program.<Main>(String[] args)

```

**Comment by mausch** [comment]
Hi, the published docker image was pretty old. I've just refreshed it, pull and try again.
At the very least you should get a more descriptive exception.

**Comment by ebertin42** [comment]
Hey it's seems that this issue is still ongoing : 

`Unhandled exception. System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JObject' to type 'Newtonsoft.Json.Linq.JArray'.
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
   at metabase_exporter.MetabaseApi.GetAllDatabaseIds() in /sln/metabase-exporter/MetabaseApi.cs:line 239
   at metabase_exporter.MetabaseApiImport.ValidateTargetDatabaseMapping(MetabaseApi api, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 71
   at metabase_exporter.MetabaseApiImport.ValidateDatabaseMapping(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 83
   at metabase_exporter.MetabaseApiImport.Import(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 24
   at metabase_exporter.Program.Import(MetabaseApi api, Import import) in /sln/metabase-exporter/Program.cs:line 45
   at metabase_exporter.Program.Main(String[] args) in /sln/metabase-exporter/Program.cs:line 27
   at metabase_exporter.Program.<Main>(String[] args)`

**Comment by ebertin42** [comment]
For more information, it's also happening with the test docker-compose provided : 

```
test_import_1         | Unhandled exception. System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JObject' to type 'Newtonsoft.Json.Linq.JArray'.
test_import_1         |    at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
test_import_1         |    at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
test_import_1         |    at metabase_exporter.MetabaseApi.GetAllDatabaseIds() in /sln/metabase-exporter/MetabaseApi.cs:line 239
test_import_1         |    at metabase_exporter.MetabaseApiImport.ValidateTargetDatabaseMapping(MetabaseApi api, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 71
test_import_1         |    at metabase_exporter.MetabaseApiImport.ValidateDatabaseMapping(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 83
test_import_1         |    at metabase_exporter.MetabaseApiImport.Import(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 24
test_import_1         |    at metabase_exporter.Program.Import(MetabaseApi api, Import import) in /sln/metabase-exporter/Program.cs:line 45
test_import_1         |    at metabase_exporter.Program.Main(String[] args) in /sln/metabase-exporter/Program.cs:line 27
test_import_1         |    at metabase_exporter.Program.<Main>(String[] args)
```

**Comment by mausch** [comment]
I'm going to need a repro, or at least more information:

- Version of Metabase you're using
- Output of `/api/database` for your instance of Metabase.

> For more information, it's also happening with the test docker-compose provided :

It works on github actions: https://github.com/elevate/elevate.metabase.tools/actions
How are you running this exactly? Try running the `run-tests.sh` script in this repo (you need Nix and Docker)

**Comment by yinshangwei** [comment]
I have the same issue.

```
Unhandled exception. System.InvalidCastException: Unable to cast object of type 'Newtonsoft.Json.Linq.JObject' to type 'Newtonsoft.Json.Linq.JArray'.
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
   at metabase_exporter.MetabaseApi.GetAllDatabaseIds() in /sln/metabase-exporter/MetabaseApi.cs:line 239
   at metabase_exporter.MetabaseApiImport.ValidateTargetDatabaseMapping(MetabaseApi api, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 71
   at metabase_exporter.MetabaseApiImport.ValidateDatabaseMapping(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 83
   at metabase_exporter.MetabaseApiImport.Import(MetabaseApi api, MetabaseState state, IReadOnlyDictionary`2 databaseMapping) in /sln/metabase-exporter/MetabaseApiImport.cs:line 24
   at metabase_exporter.Program.Import(MetabaseApi api, Import import) in /sln/metabase-exporter/Program.cs:line 45
   at metabase_exporter.Program.Main(String[] args) in /sln/metabase-exporter/Program.cs:line 27
   at metabase_exporter.Program.<Main>(String[] args)
```

**Comment by mausch** [comment]
Three different people have posted here now. Surely I can get some repro information from at least one of you? :slightly_smiling_face:

**Comment by alxp7a** [comment]
Sorry, I've given up for now.

I save the configuration files

De : Mauricio Scheffer ***@***.***>
Envoyé : mercredi 15 juin 2022 00:27
À : elevate/elevate.metabase.tools ***@***.***>
Cc : MODESTO, ALEXANDRE ***@***.***>; Author ***@***.***>
Objet : Re: [elevate/elevate.metabase.tools] cannot import backup (Issue #26)


Three different people have posted here now. Surely I can get some repro information from at least one of you? 🙂

**Comment by avshapoval** [comment]
Had the same problem, running release version without Docker (simply from command line). Solved it by commenting the whole function in MetabaseApiImport.cs.
Sure, not the best solution but it definetely works for me.
This function now: 


static async Task ValidateTargetDatabaseMapping(this MetabaseApi api, IReadOnlyDictionary<DatabaseId, DatabaseId> databaseMapping)
        {

            /*
            var databaseIds = await api.GetAllDatabaseIds();
            var incorrectMappings = databaseMapping.Where(kv => databaseIds.Contains(kv.Value) == false).ToList();
            if (incorrectMappings.Count > 0)
            {
                throw new Exception("Mappings referencing invalid databases: " +
                    string.Join(", ", incorrectMappings.Select(kv => $"{kv.Key} -> {kv.Value}")));
            }
            */

 }

**Comment by amir-dh** [comment]
I had the same problem and fix it on the https://github.com/elevate/elevate.metabase.tools/pull/31 
@mausch Please merge it.

**Comment by mausch** [comment]
Fixed for Metabase v0.40.x in 75100bc59bf0820b6c472ceebbc9b7f133a42325
